pkg/storage/crd/install_types_test.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crd
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	crdv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	crdclientfake "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/fake"
+	core "k8s.io/client-go/testing"
+)
+
+func setup(getFn, createFn func(core.Action) (bool, runtime.Object, error)) *crdclientfake.Clientset {
+	fakeClientset := &crdclientfake.Clientset{}
+
+	fakeClientset.AddReactor("get", "customresourcedefinitions", getFn)
+
+	fakeClientset.AddReactor("create", "customresourcedefinitions", createFn)
+
+	return fakeClientset
+}
+
+//make sure all resources types are installed
+func TestInstallTypesAllResources(t *testing.T) {
```

Comment:
Tests can be run concurrently to detect data races with global state. Consider adding `t.Parallel()` here and in other tests.